awx/main/tasks.py

```diff
@@ -678,31 +677,13 @@ def should_use_proot(self, instance, **kwargs):
         return False
 
     def build_inventory(self, instance, **kwargs):
-        plugin = self.get_path_to('..', 'plugins', 'inventory', 'awxrest.py')
-        if kwargs.get('isolated') is True:
-            # For isolated jobs, we have to interact w/ the REST API from the
-            # controlling node and ship the static JSON inventory to the
-            # isolated host (because the isolated host itself can't reach the
-            # REST API to fetch the inventory).
-            path = os.path.join(kwargs['private_data_dir'], 'inventory')
-            if os.path.exists(path):
-                return path
-            awxrest = imp.load_source('awxrest', plugin)
+        path = os.path.join(kwargs['private_data_dir'], 'inventory')
+        if not os.path.exists(path):
             with open(path, 'w') as f:
-                buff = cStringIO.StringIO()
-                awxrest.InventoryScript(**{
-                    'base_url': settings.INTERNAL_API_URL,
-                    'authtoken': instance.task_auth_token or '',
-                    'inventory_id': str(instance.inventory.pk),
-                    'list': True,
-                    'hostvars': True,
-                }).run(buff)
-                json_data = buff.getvalue().strip()
-                f.write("#! /usr/bin/env python\nprint '''%s'''\n" % json_data)
+                json_data = json.dumps(instance.inventory.get_script_data(hostvars=True))
+                f.write("#! /usr/bin/env python\n# -*- coding: utf-8 -*-\nprint '''%s'''\n" % json_data)
```

Comment:
Is this always a valid JSON representation? this is very interesting how you're just writing the data with JSON

Reply:
d'oh, by this point is is a JSON string, not a dict as I was assuming